src/oauth/consumer.cr

```diff
@@ -63,7 +63,8 @@ class OAuth::Consumer
                  @request_token_uri : String = "/oauth/request_token",
                  @authorize_uri : String = "/oauth/authorize",
                  @access_token_uri : String = "/oauth/access_token",
-                 @tls : Bool | OpenSSL::SSL::Context::Client = @scheme == "https")
+                 @tls : Bool | OpenSSL::SSL::Context::Client = false)
+    @tls = @scheme == "https" unless @tls.is_a?(OpenSSL::SSL::Context::Client)
```

Comment:
@sempervictus The hack you mentioned is much better written as

```cr
    tls : Nil | OpenSSL::SSL::Context::Client | Bool = nil)
  @tls = tls.nil? ? @scheme == "https" : tls
```

and that's what I'd send in as my fix, at least until this compiler bug gets fixed.